### DISCUSSION TITLE
How to use the web UI `./rspamd/#scan`?

### DISCUSSION THREAD

**Discussion by nagmat84** [rspamd/rspamd#5031] [Q&A]
Is there any explanation how to use the web UI `./rspamd/#scan`? I looked around at the official documentation site, but did not found anything. A pointer to a user guidance would be nice (if such a thing exists).

If there is no such thing, I am wondering how all those buttons and options are supposed to work together. Until now, I always only pasted raw mail message into the message source box and clicked "Upload SPAM". It seems to work, but there are further questions:

 - What is the "Scan message" button for? What it is purpose, what does it affect?
 - What is the numeric input before "Compute fuzzy hashes" for? What does it affect?
 - What is the "Compute fuzzy hashes" button for? After I have pressed it, two hash values appear at the bottom of the page, but what is the purpose of those hashes?
 - There is an area on the page called "Learn Fuzzy storage". It contains an input box "flag", an input box for "weight" and a button "upload fuzzy". How do they work?

**Comment by G2G2G2G** [comment]
No there is no explanation. I've been wondering what the exact process is for years. I think it's so simple that it seems like it'd be more complex?

- the scan message is scanning that message and it shows you it's stats on the bottom. If you scan it before clicking "upload spam" and then click scan again you will see that the "BAYES_SPAM (4.5) [100.00%]" should increase (maybe not to 100% but it'll increase)
- There is info on the fuzzy hashes https://rspamd.com/doc/tutorials/fuzzy_storage.html I don't know these really so can't answer as mine just gives errors constantly when trying to use it
